feat: add version tag to process diagram header

pedesen opened the pull request:
## Description

- move diagram header into separate component
- change diagram header to a two row layout
- add version tag to diagram header

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21185

pedesen commented:
I'll update the visual regression snapshots after the design review.

pedesen commented:
@gastonpillet01 Please check using this image: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-21185-version-tag-instances-list`. For design review purposes: The version tag is only visible if you select "Complex process" in the filter panel.

gastonpillet01 commented:
I see that text splitted into 2 rows are closer than in the instance view, and also the height is different.
I recorded a short video showing it.

Uploading 2 row header 2024-08-23 144248.mp4…

pedesen commented:
@gastonpillet01 I changed the styles to have the correct spacing. Please check again (same image name).

gastonpillet01 reviewed (approved):
Version tag correctly added to process diagram header and is also consistent with Instance view.

vitorwtavares commented on operate/client/src/App/Processes/ListView/DiagramPanel/DiagramHeader/index.tsx:
```diff
@@ -0,0 +1,103 @@
+      processName,
+      bpmnProcessId,
+      version,
+      // TODO: revert
```
Not sure what this is for, but just a reminder to not forget it.

pedesen replied:
It was for design review purposes. I reverted it.

vitorwtavares commented on operate/client/src/App/Processes/ListView/DiagramPanel/DiagramHeader/index.tsx:
```diff
@@ -0,0 +1,103 @@
+import {ProcessOperations} from '../../ProcessOperations';
+import {Restricted} from 'modules/components/Restricted';
+import {processesStore} from 'modules/stores/processes/processes.list';
+import {PanelHeader, Dd, Dl, Dt} from './styled';
```
What do you think about changing `Dd`, `Dl` and `Dt` to more descriptive names? I can infer, but without context of the feature it might make legibility a bit harder in the future.

pedesen commented:
@vitorwtavares could you please review again. I changed a few things:

- made dl, dt and dd tags more descriptive (commit https://github.com/camunda/camunda/pull/21418/commits/3d4418051ded990a3a3409c702b0ba04bf532a53)
- updated visual regression snapshots (commit https://github.com/camunda/camunda/pull/21418/commits/0e15807529ab6c9eb4821991c4544fe56c989898)
- fixed a bug where I removed the separator next to copiable content. But it should only be removed for processes, not for decisions (commit https://github.com/camunda/camunda/pull/21418/commits/7d1c548a2d21a794f2ecdb644dcdb0c0bf71bfc8). Several visual regression tests caught this:

![image](https://github.com/user-attachments/assets/e50a4136-b18f-4615-91de-e95b6bd2d81d)
